daemon/src/db.rs

```diff
@@ -249,6 +249,73 @@ async fn load_latest_cfd_state(
     Ok(latest_cfd_state_in_db)
 }
 
+pub async fn load_cfd_by_order_id(
+    order_id: OrderId,
+    conn: &mut PoolConnection<Sqlite>,
+) -> anyhow::Result<Cfd> {
+    let order_uuid = serde_json::to_string(&order_id)?;
+
+    let row = sqlx::query!(
+        r#"
+        select
+            cfds.id as cfd_id,
+            orders.uuid as order_id,
+            orders.initial_price as initial_price,
+            orders.leverage as leverage,
+            orders.trading_pair as trading_pair,
+            orders.position as position,
+            orders.origin as origin,
+            orders.liquidation_price as liquidation_price,
+            orders.term as term,
+            cfds.quantity_usd as quantity_usd,
+            cfd_states.state as state
+        from cfds as cfds
+        inner join orders as orders on cfds.order_uuid = ?
+        inner join cfd_states as cfd_states on cfd_states.cfd_id = cfds.id
+        where cfd_states.state in (
+            select
+              state
+              from cfd_states
+            where cfd_id = cfds.id
+            order by id desc
+            limit 1
+        )
+        "#,
+        order_uuid
+    )
+    .fetch_one(conn)
+    .await?;
+
+    let order_id = serde_json::from_str(row.order_id.as_str()).unwrap();
+    let initial_price = serde_json::from_str(row.initial_price.as_str()).unwrap();
+    let leverage = Leverage(row.leverage.try_into().unwrap());
+    let trading_pair = serde_json::from_str(row.trading_pair.as_str()).unwrap();
+    let liquidation_price = serde_json::from_str(row.liquidation_price.as_str()).unwrap();
+    let quantity = serde_json::from_str(row.quantity_usd.as_str()).unwrap();
+    let latest_state = serde_json::from_str(row.state.as_str()).unwrap();
+    let term = serde_json::from_str(row.term.as_str()).unwrap();
+
+    let origin: Origin = serde_json::from_str(row.origin.as_str()).unwrap();
+    let position: Position = serde_json::from_str(row.position.as_str()).unwrap();
+
+    let position = match origin {
+        Origin::Ours => position,
+        Origin::Theirs => position.counter_position(),
+    };
```

Comment:
This fucks with my head and I had to think this through twice 😅

We store an `order` with `position::sell` and load it (still has `position::sell`) and then return a `cfd` with `position::buy` if we are the taker. 

Imho the struct `Cfd` deserves some rust doc on this field or rename it to `own_position`, otherwise I can guarantee you that I will confuse things sometimes in the future.

Reply:
The `Cfd` could also be refactored to just contain the `Order` - that might make it more clear. At the moment the `Order` is flattened out in the `Cfd` which is confusing because we have to set it when loading. If we would know the complete `Order` (including `Origin`) in the `Cfd` then the `position` would always just be determined like this... 
I will see to get that together later.

Reply:
https://github.com/comit-network/hermes/pull/82